src/session/session_api.c

```diff
@@ -2219,6 +2219,38 @@ __session_get_last_error(WT_SESSION *wt_session, int *err, int *sub_level_err, c
     *err_msg = session->err_info.err_msg;
 }
 
+/*
+ * __session_set_last_error --
+ *     WT_SESSION->set_last_error method.
+ */
+static int
+__session_set_last_error(WT_SESSION *wt_session, int err, int sub_level_err)
+{
+    WT_DECL_RET;
+    WT_SESSION_IMPL *session;
+
+    session = (WT_SESSION_IMPL *)wt_session;
```

Comment:
Should we do this now @jeremythorp ?

```suggestion
    WT_SESSION_IMPL *session = (WT_SESSION_IMPL *)wt_session;
```

Reply:
Is this purely a style issue? Almost every other API in the file separates initialization & assignment, I'm wondering what the reasoning is for/against (see `__session_get_last_error` vs `__session_get_rollback_reason`).

Reply:
Yes purely a style nit pick - sorry I should have started with that. It's totally fine the way it is!
I saw another PR where @jeremythorp commented on something similar so I was just wondering.

Reply:
All good, I'll double check what convention to use, whatever it is I'll apply to both functions for consistency.

Reply:
Great question - we recently updated the WiredTiger C coding standard to allow and strongly encourage variables to be declared both with shorter scopes and to be initialised when declared. So, when writing new code, or (optionally) when modifying code, please use the new style. See https://jira.mongodb.org/browse/WT-13717.

Previously our coding standard required that variables be declared only at the start of functions and that initialisation (via assignment) was separate from declaration.

Reply:
@LX6T , what you're seeing is a lot of code that uses the old style.

Reply:
Changed in [Added assert to check for valid sub_level_err, formatting changes](https://github.com/wiredtiger/wiredtiger/pull/11350/commits/f0f443ad444cf6fe5f2fa02603569cd679c53dd6)

Reply:
I will let you resolve this thread @jeremythorp if you are happy with the update. (I am!)